adr/0006-reviewer-responsibilies.adoc

```diff
@@ -0,0 +1,74 @@
+* Being a perfect solution is _not_ a goal, _better_ than today is.
+* Define ways to measure and improve the time to review.
+
+== Scenarios (optional)
+
+List of scenarios, each described using 2-3 sentences, to capture where the ADR has an impact.
+
+== Considered options
+
+The main alternative is the status quo which roughly is: Guillaume knowing what we are intending to do and some persons talking with Guillaume regularly being sufficiently aware.
+
+== Decision
+
+There are three levels of reviewing.
+
+. Any one can review a Pull Request (PR) and give constructive feedback on different areas (code, documentation, feature experience, security, etc.).
+  This is an optional but recommended step.
+  More eyes means higher quality.
+. A PR is reviewed by a _binding reviewer_ who is fluent in the area being touched: this person is responsible for signing with blood of the usefulness, quality of the code and soundness of the solution.
+  That’s the review of the expert and the most critical one for quality.
+  This review is mandatory.
+.. Unfortunately, we do not always have (designed or effective) experts in all areas so it’s a constant work in progress and might make this step optional in some situations.
+. A PR is reviewed by a committer (push rights) whose responsibilities are:
+.. Do a general review (often from a non expert eye).
+.. Ensure consistency and quality across Quarkus’s codebase (code, concepts, API consistency, documentation).
+.. Detect potential security concerns with an attacker mindset (though not being an actual security professional).
+.. Ensure consistency at the platform level (duplicated or overlapping features, usage integration issues across features etc).
+.. Effectively push the code to the main branch.
+
+=== TODO Open questions to resolve before closing
+
+How is a person moving from casual reviewer to binding reviewer?
+People will tend to request review from binding reviewers leading to a progression problem where casual reviewers will never have the knowledge to be “promoted”.
+We have a few people with no specific area of ownership, that’s specifically acute there.
+
+How can a non binding review see the value of its work considering someone else has to go behind them anyways.
+Feels like double work where the team is very busy.
+
+How can we define SLO for the PR review to help with the _flow_, satisfy contributors while keeping quality high.
+And how to use it as a mechanism to improve and know where to get help?
+The faster the flow, the high an organisation measure in effectiveness, happiness etc.
+
+==== Proposition 1
+
+Define and measure (GitHub APIs) SLO for the time to push or drop / withdraw.
+This is not a perfect proxy as some PRs do demand multi weeks conversations but measuring and dashboarding is our first step to know in which area we have a problem.
+
+We could define a team intended SLO (say 3 weeks to start but that’s already quite long) and know which areas chronically fall behind.
```

Comment:
Three is not that long. 

There is also another issue. Let's imagine 2 PRs are opened at the same time. One gets expedited quickly and they other one is staled. There are multiple reasons for this:

- we often trust more regular contributors - that isn't very objective, but that's the reality
Some PRs are more complex to review than others. Changing the native compilation configuration can be pretty touchy and requires the reviewer to test many possibilities. 
- some PR requires expertise even binding reviewers do not have (like Windows :-)) - cross-cutting PRs can also be tricky (because it requires multiple binding reviews, and technical friction may happen)

Reply:
Yes, that's where it's important to have the conversation.

* what are the useful SLO (is that 90% percentile vs absolute number)
* is a latency between last user comment and committer and binding reviewer comment useful (e.g. 95% before 7 days)

And even before defining the SLO, probably defining and measuring the SLI could be useful as an indicator.